From 8bcc975327e9152641eaf7813f7ed4a292c060b7 Mon Sep 17 00:00:00 2001
From: SeongJae Park <sj@kernel.org>
Date: Sun, 4 Aug 2024 11:49:42 -0700
Subject: [PATCH] Revert "mm/damon/Kconfig: select DAMON modules to test when
 test config is set"

This reverts commit 142c772af333ce3a5d488eecb912ed006a6e49c4.

The commit may abusing 'select' too much.  Actually it caused some
regression reports[1].  Maybe adding .kunitconfig file for DAMON under
mm/damon/tessts/ like other tests  (e.g., ./drivers/clk/.kunitconfig)is
the right approach.  Revert the commit for now.

[1] https://lore.kernel.org/202408040440.Oe31R40L-lkp@intel.com

Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202408040440.Oe31R40L-lkp@intel.com/
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/Kconfig | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/mm/damon/Kconfig b/mm/damon/Kconfig
index 996c8ff67c8d..4fe7520601dd 100644
--- a/mm/damon/Kconfig
+++ b/mm/damon/Kconfig
@@ -14,8 +14,7 @@ config DAMON
 
 config DAMON_KUNIT_TEST
 	bool "Test for damon" if !KUNIT_ALL_TESTS
-	depends on KUNIT=y
-	select DAMON
+	depends on DAMON && KUNIT=y
 	default KUNIT_ALL_TESTS
 	help
 	  This builds the DAMON Kunit test suite.
@@ -43,8 +42,7 @@ config DAMON_PADDR
 
 config DAMON_VADDR_KUNIT_TEST
 	bool "Test for DAMON operations" if !KUNIT_ALL_TESTS
-	depends on KUNIT=y
-	select DAMON_VADDR
+	depends on DAMON_VADDR && KUNIT=y
 	default KUNIT_ALL_TESTS
 	help
 	  This builds the DAMON virtual addresses operations Kunit test suite.
@@ -63,8 +61,7 @@ config DAMON_SYSFS
 
 config DAMON_SYSFS_KUNIT_TEST
 	bool "Test for damon debugfs interface" if !KUNIT_ALL_TESTS
-	depends on KUNIT=y
-	select DAMON_SYSFS
+	depends on DAMON_SYSFS && KUNIT=y
 	default KUNIT_ALL_TESTS
 	help
 	  This builds the DAMON sysfs interface Kunit test suite.
@@ -94,8 +91,7 @@ config DAMON_DBGFS
 
 config DAMON_DBGFS_KUNIT_TEST
 	bool "Test for damon debugfs interface" if !KUNIT_ALL_TESTS
-	depends on KUNIT=y
-	select DAMON_DBGFS
+	depends on DAMON_DBGFS && KUNIT=y
 	default KUNIT_ALL_TESTS
 	help
 	  This builds the DAMON debugfs interface Kunit test suite.
-- 
2.39.2

